File: Codeforces/B_Make_Equal.py
def solve():
    N = int(input())
    A = list(map(int, input().split()))
    sum_val = sum(A)
    M = sum_val // N
    B = 0
    for a in A:
        if a > M:
            B += (a - M)
        else:
            X = M - a
            B -= X
        if B < 0:
            print("NO")
            return
    print("YES")

if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        solve()

File: Codeforces/B_Make_Equal.cpp
#include <bits/stdc++.h>
//#define int long long
using namespace std;

void solve(){
int N, sum = 0; cin >> N;
vector<int> A(N); 
for(auto &a : A){
    cin >> a;
    sum += a;
}
int M = sum/N;
int B = 0;
for(int i = 0 ; i < N ; i++){
    if(A[i] > M){
        B+=(A[i] - M);
    }
    else{
        int X = M - A[i];
        B -= X;
    }
    if(B < 0){
        cout << "NO" << endl;
        return;
    }
}
cout << "YES" << endl;
}

int32_t main(){
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    int t;
    cin >> t;
    while(t--){
        solve();
    }
    return 0;
}